tests/test_build.py

```diff
@@ -25,6 +25,15 @@ def test_conditional_assets(app, docname, check_asset_links):
         )
 
 
+@pytest.mark.parametrize("docname", ["index", "no_tabs1", "no_tabs2"])
+@pytest.mark.sphinx(testroot="conditionalassets")
+@pytest.mark.skipif(
+    sphinx.version_info[:2] >= (4, 1), reason="Test uses Sphinx 4.1 config"
+)
+def test_conditional_assets(app, docname, check_asset_links):
+    check_asset_links(app, filename=docname + ".html")
```

Comment:
Test is currently failing - I think it should use the new sphinx API?

```suggestion
    app.registry.html_assets_policy == "always"
    check_asset_links(app, filename=docname + ".html")
```

Reply:
Yes. I updated this test.

However, it kept failing because it seems the output is firstly generated by `test_conditional_assets` test (without `app.set_html_assets_policy("always")`) and the result is cached.

What should I do here? Create a new `testroot=` equal to `conditionalassets`?

NOTE: in an extension that I'm a maintainer, I'm cleaning the output's path after the test is ran to avoid this problem, see https://github.com/readthedocs/sphinx-hoverxref/blob/master/tests/conftest.py#L9 --maybe something like that is a good approach here as well?

Reply:
Thanks, that looks spot on.

It looks like this is passing the CI tests now. I'm surprised though, given the issue that you describe using the same `testroot`. Is this failing locally for you?

If yes, please do create a separate root folder for this test. I like your fixture for cleaning up after tests, but do find the test outputs useful for debugging when the tests have failed (locally).

`pre-commit` is picking up on a couple of minor formatting issues below